### PR TITLE
UI Fixed issue Spaces in test suite name should not be allowed #7872

### LINKED ISSUE #7872: Spaces in test suite name should not be allowed
Screenshot 2022-10-03 at 9 13 15 AM

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

I worked on the below issue:
- Spaces in test suite name should not be allowed #7872
- Redirection issue on click of the view test suite

Closes #7872 

#
### Type of change :

- [x] Bug fix

#
### Frontend Preview (Screenshots) :

- Test suite name issue

https://user-images.githubusercontent.com/71748675/193537465-54061150-bbdf-4b76-bdcc-96c4b4b336d7.mov

- Redirection issue

https://user-images.githubusercontent.com/71748675/193538091-5d12da0f-0434-4959-b49a-9203c447c3cb.mov

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/AddDataQualityTest/components/SelectTestSuite.tsx [hunk 1]
```diff
@@ -167,6 +167,10 @@ const SelectTestSuite: React.FC<SelectTestSuiteProps> = ({
                     required: isEmpty(form.getFieldValue('testSuiteId')),
                     message: 'Name is required!',
                   },
+                  {
+                    pattern: /^[A-Za-z0-9_]*$/g,
+                    message: 'Spacial character is not allowed!',
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/AddDataQualityTest/components/SelectTestSuite.tsx:172]: Can we take this message from Json const?

**Review by Sachin-chaurasiya — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/AddDataQualityTest/components/SelectTestSuite.tsx [hunk 2]
```diff
@@ -167,6 +168,10 @@ const SelectTestSuite: React.FC<SelectTestSuiteProps> = ({
                     required: isEmpty(form.getFieldValue('testSuiteId')),
                     message: 'Name is required!',
                   },
+                  {
+                    pattern: /^[A-Za-z0-9_]*$/g,
+                    message: jsonData.label['spatial-character-error'],
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/AddDataQualityTest/components/SelectTestSuite.tsx:173]:
```suggestion
                    message: jsonData.label['special-character-error'],
```

File: openmetadata-ui/src/main/resources/ui/src/jsons/en.ts [hunk 3]
```diff
@@ -200,6 +200,7 @@ const jsonData = {
       'Once you delete this entity, it will be removed permanently.',
     'email-confirmation':
       'Please confirm your email, confirmation has been sent to your email',
+    'spatial-character-error': 'Spacial character is not allowed!',
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/jsons/en.ts:203]:
```suggestion
    'spacial-character-error': 'Spacial character is not allowed!',
```

**Comment by chirag-madlani** [comment]
@ShaileshParmar11 merge with main to fix maven CI failure

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**